### PR TITLE
6X: Upgrade resource queues and groups

### PR DESCRIPTION
**Pull request by soumyadeep2007**
This PR consists of two commits. Please review these separately.

1) Fix dump of built-in resource groups.
2) Pass `--resource-groups` and `--resource-queues` to `pg_dumpall` during a `pg_upgrade`.

Pipeline: https://cm.ci.gpdb.pivotal.io/teams/main/pipelines/dumpResgrp_6X

Upgradeable regress-style tests (which tests the schema upgrade): https://github.com/greenplum-db/gpupgrade/pull/561

Additional testing: This PR was tested in a CI environment supporting resource groups by
upgrading a resource group from `resgroup_memory_limit.source` and checking that
the memory limits prescribed by that group were respected when queries on the target
cluster exceeded the limits.

Co-authored-by: Kalen Krempely <kkrempely@vmware.com>

### CODE REVIEW

File: src/bin/pg_dump/pg_dumpall.c [hunk 1]
```diff
@@ -770,28 +769,31 @@ dumpResGroups(PGconn *conn)
 
 		resetPQExpBuffer(buf);
 
-		/* DROP or CREATE default group, so ALTER it  */
 		if (0 == strcmp(groupname, "default_group") || 0 == strcmp(groupname, "admin_group"))
```
Comment (zhangh43) [inline src/bin/pg_dump/pg_dumpall.c:772]: Why do we remove `fmtId` in `groupname = fmtId(PQgetvalue(res, i, i_groupname));`. It seems that "default_group" and "admin_group" will not introduce additional quote in `fmtId`.

File: src/bin/pg_dump/pg_dumpall.c [hunk 2]
```diff
@@ -743,10 +742,10 @@ dumpResGroups(PGconn *conn)
 	 * total cpu_rate_limit and memory_limit should less than 100, so clean
 	 * them before we seting new memory_limit and cpu_rate_limit.
 	 */
-	fprintf(OPF, "ALTER RESOURCE GROUP admin_group SET cpu_rate_limit 1;\n");
-	fprintf(OPF, "ALTER RESOURCE GROUP default_group SET cpu_rate_limit 1;\n");
-	fprintf(OPF, "ALTER RESOURCE GROUP admin_group SET memory_limit 1;\n");
-	fprintf(OPF, "ALTER RESOURCE GROUP default_group SET memory_limit 1;\n");
+	fprintf(OPF, "ALTER RESOURCE GROUP \"admin_group\" SET cpu_rate_limit 1;\n");
+	fprintf(OPF, "ALTER RESOURCE GROUP \"default_group\" SET cpu_rate_limit 1;\n");
```
Comment (zhangh43) [inline src/bin/pg_dump/pg_dumpall.c:746]: why do we need quote here?

File: src/bin/pg_dump/pg_dumpall.c (hunk 1 above)
Comment (soumyadeep2007) [reply src/bin/pg_dump/pg_dumpall.c:772]: Earlier, in this [line](https://github.com/greenplum-db/gpdb/pull/12143/files#diff-a9d1d4cd0a7d5904dc1d75fbe61095c0214efa3bc2e3c3abc59833add30a3a20L762), we were calling `fmtId` which was causing `groupname` to have quotes in it.

This is why the comparison `(0 == strcmp(groupname, "default_group"))` was false even though when groupname was `default_group`

Actually, the above conditions would only be true if we wrote the condition as follows:
```c
if (0 == strcmp(groupname, "\"default_group\"") || 0 == strcmp(groupname, "\"admin_group\""))
```

That was messy, so we decided to compare on the basis of the raw value: `PQgetvalue(res, i, i_groupname)` and only use `fmtId` when emitting SQL statements to the file.

File: src/bin/pg_dump/pg_dumpall.c (hunk 2 above)
Comment (soumyadeep2007) [reply src/bin/pg_dump/pg_dumpall.c:746]: Cosmetic reason: For a consistent dump. `admin_group` and `default_group` would be quoted in other statements in the dump file.

**Review by zhangh43 — APPROVED** [review]

**Pull request merged**